apps/web/src/services/safe-apps/manifest.ts

```diff
@@ -98,10 +103,14 @@ const fetchSafeAppFromManifest = async (
 
   const iconUrl = getAppLogoUrl(normalizedAppUrl, appManifest)
 
+  // Preserve the original URL with parameters
+  const originalUrl = appUrl
+
   return {
     // Must satisfy https://docs.djangoproject.com/en/5.0/ref/models/fields/#positiveintegerfield
     id: Math.round(Math.random() * 1e9 + 1e6),
-    url: normalizedAppUrl,
+    url: normalizedAppUrl, // Store the base URL without parameters for matching
+    originalUrl, // Store the original URL with parameters
```

Comment:
No need to create a separate variable, you can pass appUrl directly.

```suggestion
    originalUrl: appUrl, // Store the original URL with parameters
```